core/src/main/java/com/oath/oak/OakMap.java

```diff
@@ -391,59 +426,470 @@ public boolean inBounds(K key) {
                 this.fromKey, this.fromInclusive, this.toKey, this.toInclusive, true, threadIndexCalculator);
     }
 
+    @Override
+    public NavigableSet<K> navigableKeySet() {
+        return new KeySet<>(this);
+    }
+
+    @Override
+    public NavigableSet<K> keySet() {
+        return new KeySet<>(this);
+    }
+
+    @Override
+    public NavigableSet<K> descendingKeySet() {
+        return descendingMap().navigableKeySet();
+    }
+
+    @Override
+    public Set<Entry<K, V>> entrySet() {
+        return new EntrySet<>(this);
+    }
+
+    @Override
+    public Collection<V> values() {
+        return new Values<>(this);
+    }
+
+    /* ------ Zero-Copy API methods  ------ */
+
+    public ZeroCopyMap ZC() {
+        return new ZeroCopyMap(this);
+    }
+
+    public static class ZeroCopyMap<K, V>  {
```

Comment:
Can this class move to a separate file? and anyway it shouldnt be in the middle of the class - this is very confusing
Shouldn't this extend OakMap or NavigableSomething? Or maybe have a common ancestor for both ZCMap and OakMap?

Reply:
We can move this to a separate file and lose some type safety, since the type parameters K and V will not be bound. 
There are no common ancestors since our zero-copy implementation make some assumptions about in/out arguments to some methods that are in conflict with standard interfaces (for example, we use bytebuffers instead of keys/values in certain places).